kratos/utilities/geometry_tester.cpp

```diff
@@ -944,8 +964,74 @@ bool GeometryTesterUtility::VerfiyShapeFunctionsValues(
 /***********************************************************************************/
 /***********************************************************************************/
 
+bool GeometryTesterUtility::VerifyShapeFunctionsSecondDerivativesValues(
+    GeometryType& rGeometry,
+    GeometryType::CoordinatesArrayType& rGlobalCoordinates,
+    std::stringstream& rErrorMessage
+    )
+{
+    GeometryType::CoordinatesArrayType local_coordinates;
+    rGeometry.PointLocalCoordinates( local_coordinates, rGlobalCoordinates );
+
+    Vector f_1 = ZeroVector(rGeometry.size());
+    Vector f_2 = ZeroVector(rGeometry.size());
+    Vector f_3 = ZeroVector(rGeometry.size());
+    Vector f_4 = ZeroVector(rGeometry.size());
+
+    DenseVector<Matrix> DDN_DX, H;
+    Vector ei,ej,f;
+
+    double delta = 1e-1;
+    unsigned int dim = rGeometry.Dimension();
+    rGeometry.ShapeFunctionsSecondDerivatives(DDN_DX,local_coordinates);
+
+    if ( H.size() != rGeometry.size() )
+    {
+        // KLUDGE: While there is a bug in
+        // ublas vector resize, I have to put this beside resizing!!
+        DenseVector<Matrix> temp( rGeometry.size() );
+        H.swap( temp );
+    }
+
+
+    for ( unsigned int i = 0; i < rGeometry.size(); ++i ) H[i].resize(dim, dim, false);
+
+    for (unsigned int i = 0; i<dim;i++){
+        for (unsigned int j = 0; j<dim;j++){
+            ei = ZeroVector(dim);
+            ei[i] = 1.0;
+            ej = ZeroVector(dim);
+            ej[j] = 1.0;
+            std::transform(ei.begin(), ei.end(), ei.begin(), [delta](double &c){ return c*delta; });
+            std::transform(ej.begin(), ej.end(), ej.begin(), [delta](double &c){ return c*delta; });
+            rGeometry.ShapeFunctionsValues(f_1,local_coordinates + ei + ej);
+            rGeometry.ShapeFunctionsValues(f_2,local_coordinates + ei - ej);
+            rGeometry.ShapeFunctionsValues(f_3,local_coordinates - ei + ej);
+            rGeometry.ShapeFunctionsValues(f_4,local_coordinates - ei - ej);
+            f = f_1-f_2-f_3+f_4;
+            transform(f.begin(), f.end(), f.begin(), [delta](double &c){ return c/(4.0*std::pow(delta,2)); });
```

Comment:
```suggestion
           std::transform(f.begin(), f.end(), f.begin(), [delta](double &c){ return c/(4.0*std::pow(delta,2)); });
```

Reply:
I am not sure if this will work since this modification was included in some commits before. However, I will apply it.

Reply:
To be honest no idea, but it looks to me like if it is getting the `transform` definition from somewhere outside this class (in the CI the unity built is enabled so it may have access to definitions it normally should not)

In any case is worth taking a look just to confirm that is doing what it was suppose to.